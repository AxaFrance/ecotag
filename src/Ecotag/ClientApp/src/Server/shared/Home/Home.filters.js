export const filterPaging = (items, numberItems, currentPage) => {
    let pageIndex = 0;
    if (currentPage > 1) {
        pageIndex = currentPage - 1;
    }

    let controlledNumberItems = 0;
    if (numberItems > 0) {
        controlledNumberItems = numberItems;
    }

    const length = items.length;
    const beginIndex = pageIndex * controlledNumberItems;
    const lastIndex = length < beginIndex + controlledNumberItems ? length : beginIndex + controlledNumberItems;

    return items.slice(beginIndex, lastIndex);
};

export const computeNumberPages = (items, numberItems) => {
    if (!items) {
        return 1;
    }
    const numberItemsControlled = numberItems < 1 ? 1 : numberItems;
    const length = items.length;
    return length > 0 ? Math.ceil(length / numberItemsControlled) : 1;
};

export const getItemsFiltered = (items, filterValue) => {
    if (filterValue !== null && filterValue !== undefined && filterValue.length > 0) {
        return items.filter(project => project.name.toLowerCase().includes(filterValue.toLowerCase()));
    }
    return items;
};

export const sortByTimeLastUpdate = (a, b) => {
    if (a.value.timeLastUpdate === b.value.timeLastUpdate) {
        return 0;
    }

    if (a.value.timeLastUpdate === null) {
        return 1;
    }

    if (b.value.timeLastUpdate === null) {
        return -1;
    }

    return a.value.timeLastUpdate - b.value.timeLastUpdate;
};

const computeEntries = columns => {
    const entries = [];
    for (const [key, value] of Object.entries(columns)) {
        entries.push({key, value});
    }
    return entries;
};

const compareItems = (entry, itemA, itemB) => {
    if (typeof itemB[entry.key] === 'number') {
        return itemB[entry.key] < itemA[entry.key] ? 1 : -1;
    }
    if (itemB[entry.key] && itemA[entry.key]) {
        if (itemB[entry.key] instanceof Date) {
            return itemA[entry.key].getTime() - itemB[entry.key].getTime();
        }
        const localComp = itemA[entry.key].localeCompare(itemB[entry.key]);
        if (localComp !== 0) {
            return localComp;
        }
    }
};

const compareItemsBySortingType = (columns, entry, itemA, itemB) => {
    if (columns[entry.key].value === 'asc') {
        return compareItems(entry, itemA, itemB);
    }
    if (columns[entry.key].value === 'desc') {
        return compareItems(entry, itemB, itemA);
    }
};

export const getItemsSorted = (items, columns) => {
    const entries = computeEntries(columns);
    entries.sort(sortByTimeLastUpdate);
    const criteria = entries[0];
    return [...items].sort((itemA, itemB) => compareItemsBySortingType(columns, criteria, itemA, itemB));
};
