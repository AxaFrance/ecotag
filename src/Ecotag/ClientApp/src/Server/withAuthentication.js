import React from 'react';
import {useOidcAccessToken, useOidcIdToken, useOidcUser} from '@axa-fr/react-oidc';
import {EnvironmentConsumer} from './EnvironmentProvider';

const NON_CONNECTE = 'Non ConnectÃ©';

const getAuthName = (dataArray, propertyName = "name", defaultResult = NON_CONNECTE) => {

    for (let i = 0; i < dataArray.length; i++) {
        const item = dataArray[i];
        if (item && item[propertyName]) {
            return item[propertyName];
        }
    }

    return defaultResult;
}

export const DataScientist = "ECOTAG_DATA_SCIENTIST";
export const Annotateur = "ECOTAG_ANNOTATEUR";
export const Administateur = "ECOTAG_ADMINISTRATEUR";

const addRole = (roles, role) => {
    if (roles.includes(role)) {
        return;
    }
    if (role === DataScientist || role === Annotateur || role === Administateur) {
        roles.push(role);
    }
}

export const extractRoles = (accessTokenPayload, oidcMode) => {
    const roles = [];
    if (oidcMode === "AXA_FRANCE") {
        if (accessTokenPayload && accessTokenPayload.member_of && accessTokenPayload.member_of.length > 0) {
              accessTokenPayload.forEach((item) => addRole(roles, item));
            if (roles.includes(DataScientist)) {
                addRole(roles, Annotateur);
            } else if (roles.includes(Administateur)) {
                addRole(roles, DataScientist)
                addRole(roles, Annotateur)
            }
        }
    } else {
        roles.push(Annotateur);
        roles.push(DataScientist);
        roles.push(Administateur);
    }
    return roles;
}

const extractDataFromOAuthToken = (idTokenPayload, accessTokenPayload, oidcUser, environment) => ({
    name: getAuthName([idTokenPayload, accessTokenPayload]),
    email: getAuthName([oidcUser, accessTokenPayload], 'email', ''),
    roles: extractRoles(accessTokenPayload, environment.oidc.mode)
});

const withAuthentication = () => Component => props => {
    const {idTokenPayload} = useOidcIdToken();
    const {accessTokenPayload} = useOidcAccessToken();
    const {oidcUser, oidcUserLoadingState} = useOidcUser();
    return <EnvironmentConsumer>{store => <Component {...props}
                                                     user={extractDataFromOAuthToken(idTokenPayload, accessTokenPayload, oidcUser, store.environment)}
                                                     UserLoadingState={oidcUserLoadingState}/>}</EnvironmentConsumer>;
};

export default withAuthentication;
