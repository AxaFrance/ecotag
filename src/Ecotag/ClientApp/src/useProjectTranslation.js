import './i18n';
import {useTranslation} from "react-i18next";
import i18next from 'i18next';

const useProjectTranslation = (namespace = 'translation') => {
    const {t: translate} = useTranslation(namespace);
    return {translate};
};

export const changeProjectTranslationLanguage = (language) => {
    i18next.changeLanguage(language);
};

export default useProjectTranslation;
