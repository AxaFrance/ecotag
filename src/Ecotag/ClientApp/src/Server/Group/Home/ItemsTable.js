import React, {useState} from "react";
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import Action from '@axa-fr/react-toolkit-action';
import Edit from "./Edit";
import './ItemsTable.scss';
import useProjectTranslation from "../../../useProjectTranslation";

const computeUsersList = users => {
    if (users.length === 0) {
        return "-";
    }
    return users.map(user => user.email).join(', ');
};

const UserRow = ({id, name, users, eligibleUsers, onUpdateUser}) => {
    const [isManageUsersModalVisible, setManageUsersModalVisible] = useState(false);
    const {translate} = useProjectTranslation();
    return (
        <Table.Tr key={id}>
            <Table.Td>{name}</Table.Td>
            <Table.Td>{computeUsersList(users)}</Table.Td>
            <Table.Td class="af-table__cell af-table__small">
                <Action id="editActionId" icon="pencil" title={translate('group.list.edit')}
                        onClick={() => setManageUsersModalVisible(true)}/>
                <Edit
                    idGroup={id}
                    users={users}
                    eligibleUsers={eligibleUsers}
                    onUpdateUser={onUpdateUser}
                    isManageUsersModalVisible={isManageUsersModalVisible}
                    setManageUsersModalVisible={setManageUsersModalVisible}
                />
            </Table.Td>
        </Table.Tr>
    );
};

const ItemsTable = ({items, filters, onChangePaging, onUpdateUser}) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.name')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.users')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.actions')}</span>
                        </Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(({id, name, users = [], eligibleUsers = []}) => (
                        <UserRow
                            key={id}
                            id={id}
                            name={name}
                            users={users}
                            eligibleUsers={eligibleUsers}
                            onUpdateUser={onUpdateUser}
                        />
                    ))}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
                displayLabel={translate('group.paging.display')}
                elementsLabel={translate('group.paging.elements')}
                previousLabel={translate('group.paging.previous')}
                nextLabel={translate('group.paging.next')}
            />
        </>
    )
};

export default ItemsTable;