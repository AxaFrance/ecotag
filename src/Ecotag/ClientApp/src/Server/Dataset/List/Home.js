import React from 'react';
import {Link} from 'react-router-dom';
import Title from '../../../TitleBar';
import './Home.scss';
import ItemsTable from "./ItemsTable";
import EmptyArrayManager from "../../../EmptyArrayManager";
import useProjectTranslation from "../../../useProjectTranslation";

const Home = ({items, numberTotalItems, filters, onChangePaging, onChangeFilter}) => {
    const {translate} = useProjectTranslation();
    return (
        <>
            <Title title={translate('dataset.title')} subtitle={translate('dataset.subtitle')}/>
            <div className="af-home container">
                <Link className="btn af-btn af-btn--quote" to="/datasets/new">
                    <span className="af-btn__text">{translate('dataset.new_dataset')}</span>
                </Link>
                <a className="af-home__access-token" target="_self" href="/access-token">{translate('dataset.access_token')}</a>
                <h1 className="af-title--content">{translate('dataset.active_datasets_counter')}{numberTotalItems}</h1>
                <div className="row row--datasets-filters">
                    <div className="col">
                        <div className="af-filter-inline">
                            <span className="af-filter-inline__title">
                                <span className="glyphicon glyphicon-filter"/>
                                <span className="af-filter-inline__title-text">{translate('dataset.filter_by')}</span>
                            </span>
                            <div className="af-filter-inline__field">
                                <label className="af-form__group-label" htmlFor="inputtext1">{translate('dataset.name')}</label>
                                <div className="af-form__text">
                                    <input className="af-form__input-text" id="inputtext1" name="inputtextname"
                                           onChange={onChangeFilter} type="text"/>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <EmptyArrayManager items={items} emptyArrayMessage={translate('dataset.empty_list')}>
                    <ItemsTable
                        items={items}
                        filters={filters}
                        onChangePaging={onChangePaging}
                        onChangeFilter={onChangeFilter}/>
                </EmptyArrayManager>
            </div>
        </>
    )
};

export default Home;
