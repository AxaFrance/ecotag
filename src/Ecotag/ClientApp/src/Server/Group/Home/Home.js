import React from 'react';
import Title from '../../../TitleBar';
import New from './New';

import './Home.scss';
import '../../shared/Modal/modal.scss';
import EmptyArrayManager from "../../../EmptyArrayManager";
import ItemsTable from "./ItemsTable";
import useProjectTranslation from "../../../useProjectTranslation";

const Home = ({
                  items,
                  numberItemsTotal,
                  filters,
                  onChangePaging,
                  fields,
                  isSubmitable,
                  hasSubmit,
                  onChangeCreateGroup,
                  onSubmitCreateGroup,
                  onUpdateUser,
              }) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Title subtitle={translate('group.subtitle')} title={translate('group.title')}/>
            <div className="af-home container">
                <h1 className="af-title--content">{translate('group.creation_title')}</h1>
                <New
                    disabled={isSubmitable}
                    fields={fields}
                    hasSubmit={hasSubmit}
                    onChangeCreateGroup={onChangeCreateGroup}
                    onSubmitCreateGroup={onSubmitCreateGroup}
                />
                <h1 className="af-title--content">
                    {`${translate('group.count_teams.first_part')}${numberItemsTotal}${translate('group.count_teams.second_part')}`}
                </h1>
                <EmptyArrayManager items={items} emptyArrayMessage={translate('group.empty_list')}>
                    <ItemsTable
                        items={items}
                        filters={filters}
                        onChangePaging={onChangePaging}
                        onUpdateUser={onUpdateUser}
                    />
                </EmptyArrayManager>
            </div>
        </>
    );
};

export default Home;
