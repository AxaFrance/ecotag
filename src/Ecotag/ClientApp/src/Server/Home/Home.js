import React from "react";
import Title from "../../TitleBar";
import {Link} from 'react-router-dom';
import './Home.scss';
import withAuthentication, {Administateur, Annotateur, DataScientist} from "../withAuthentication";
import {OidcUserStatus} from "@axa-fr/react-oidc";
import {useTranslation} from "react-i18next";

export const Home = ({user: {roles = []}, userLoadingState}) => {
    const {t} = useTranslation();

    return (
        <div className="home">
            <Title title={t('home.title')} goButton={false}/>
            <div className="home__links-container">
                {roles.includes(Annotateur) && <Link className="home__link" to="/projects">
                    <div className="home__link-container home__link-container--projects">{t('home.projects')}</div>
                </Link>}
                {roles.includes(DataScientist) && <Link className="home__link" to="/datasets">
                    <div className="home__link-container home__link-container--datasets">{t('home.datasets')}</div>
                </Link>}
                {roles.includes(Administateur) && <Link className="home__link" to="/teams">
                    <div className="home__link-container home__link-container--groups">{t('home.groups')}</div>
                </Link>}
                {(!roles.includes(Annotateur) && !roles.includes(DataScientist) && !roles.includes(Administateur) && userLoadingState === OidcUserStatus.Loaded) &&
                    <p>{t('home.errors.no_roles')}</p>}
            </div>
        </div>
    );
}

export default withAuthentication()(Home);
