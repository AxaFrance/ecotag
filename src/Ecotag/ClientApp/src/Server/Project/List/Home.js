import React from 'react';
import {Link,} from 'react-router-dom';
import Title from '../../../TitleBar';
import './Home.scss';
import '../../shared/Modal/modal.scss';
import EmptyArrayManager from '../../../EmptyArrayManager';
import ItemsTable from './ItemsTable';
import {DataScientist} from '../../withAuthentication';
import useProjectTranslation from '../../../useProjectTranslation';

const Home = ({items, numberTotalItems, filters, onChangePaging, onChangeFilter, onChangeSort, fetch, user}) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Title title={translate('project.projects_list.title')} subtitle={translate('project.projects_list.subtitle')}/>
            <div className="af-home container">
                {user.roles.includes(DataScientist) && <Link className="btn af-btn af-btn--quote" to="/projects/new">
                    <span className="af-btn__text">{translate('project.projects_list.new')}</span>
                </Link>}
                <h1 className="af-title--content">{`${translate('project.projects_list.active_projects_counter')} (${numberTotalItems})`}</h1>
                <div className="row row--projects-filters">
                    <div className="col">
                        <div className="af-filter-inline">
                            <span className="af-filter-inline__title">
                                <span className="glyphicon glyphicon-filter"/>
                                <span className="af-filter-inline__title-text">{translate('project.projects_list.filter_by')}</span>
                            </span>
                            <div className="af-filter-inline__field">
                                <label className="af-form__group-label" htmlFor="inputTextFilterProjects">
                                    {translate('project.projects_list.project_name')}
                                </label>
                                <div className="af-form__text">
                                    <input
                                        className="af-form__input-text"
                                        id="inputTextFilterProjects"
                                        name="inputTextFilterProjects"
                                        onChange={event => onChangeFilter(event.target.value)}
                                        type="text"
                                    />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <EmptyArrayManager items={items} emptyArrayMessage={translate('project.projects_list.no_elements')}>
                    <ItemsTable
                        items={items}
                        filters={filters}
                        onChangePaging={onChangePaging}
                        onChangeSort={onChangeSort}
                        fetch={fetch}
                    />
                </EmptyArrayManager>
            </div>
        </>
    );
};

export default Home;
