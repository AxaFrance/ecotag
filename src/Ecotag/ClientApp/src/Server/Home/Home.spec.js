import React from 'react';
import '@testing-library/jest-dom';
import {render, waitFor} from '@testing-library/react';
import {Home} from './Home';
import {BrowserRouter as Router} from 'react-router-dom';
import {Administateur, Annotateur, DataScientist} from '../withAuthentication';
import {OidcUserStatus} from '@axa-fr/react-oidc';
import {changeProjectTranslationLanguage} from "../../useProjectTranslation";

describe.each([
    [`${DataScientist},${Annotateur}`],
    [Annotateur],
    [`${DataScientist},${Annotateur},${Administateur}`],
    [""]
])('Home %p', (roles) => {
    test('Render home page with correct buttons depending on user roles', async () => {
        changeProjectTranslationLanguage('en');
        const user = {
            roles: roles ? roles.split(",") : [],
            name: 'Guillaume Chervet'
        }
        const {container, asFragment} = render(<Router basename="/"><Home user={user}
                                                                          userLoadingState={OidcUserStatus.Loaded}/></Router>);

        if (roles.includes(Annotateur)) {
            await waitFor(() => expect(container.querySelector('.home__link-container--projects')).not.toBeNull());
        }
        if (roles.includes(DataScientist)) {
            await waitFor(() => expect(container.querySelector('.home__link-container--datasets')).not.toBeNull());
        }
        if (roles.includes(Administateur)) {
            await waitFor(() => expect(container.querySelector('.home__link-container--groups')).not.toBeNull());
        }

        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Home translation', () => {
    const user = {
        roles: [DataScientist,Annotateur,Administateur],
        name: 'Guillaume Chervet'
    };

    it('Renders component with english translation', () => {
        changeProjectTranslationLanguage('en');
        const {asFragment} = render(<Router basename="/"><Home user={user} userLoadingState={OidcUserStatus.Loaded}/></Router>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('Renders component with french translation', () => {
        changeProjectTranslationLanguage('fr');
        const {asFragment} = render(<Router basename="/"><Home user={user} userLoadingState={OidcUserStatus.Loaded}/></Router>);
        expect(asFragment()).toMatchSnapshot();
    });
});
