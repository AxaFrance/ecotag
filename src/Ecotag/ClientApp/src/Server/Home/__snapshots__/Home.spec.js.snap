// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home "" Render home page with correct buttons depending on user roles 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <p>
        No role is associated to your profile.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home "ECOTAG_ANNOTATEUR" Render home page with correct buttons depending on user roles 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/projects"
      >
        <div
          class="home__link-container home__link-container--projects"
        >
          Projects
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home "ECOTAG_DATA_SCIENTIST,ECOTAG_ANNOTATEUR" Render home page with correct buttons depending on user roles 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/projects"
      >
        <div
          class="home__link-container home__link-container--projects"
        >
          Projects
        </div>
      </a>
      <a
        class="home__link"
        href="/datasets"
      >
        <div
          class="home__link-container home__link-container--datasets"
        >
          Datasets
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home "ECOTAG_DATA_SCIENTIST,ECOTAG_ANNOTATEUR,ECOTAG_ADMINISTRATEUR" Render home page with correct buttons depending on user roles 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/projects"
      >
        <div
          class="home__link-container home__link-container--projects"
        >
          Projects
        </div>
      </a>
      <a
        class="home__link"
        href="/datasets"
      >
        <div
          class="home__link-container home__link-container--datasets"
        >
          Datasets
        </div>
      </a>
      <a
        class="home__link"
        href="/teams"
      >
        <div
          class="home__link-container home__link-container--groups"
        >
          Groups
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home translation Renders component with english translation 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/projects"
      >
        <div
          class="home__link-container home__link-container--projects"
        >
          Projects
        </div>
      </a>
      <a
        class="home__link"
        href="/datasets"
      >
        <div
          class="home__link-container home__link-container--datasets"
        >
          Datasets
        </div>
      </a>
      <a
        class="home__link"
        href="/teams"
      >
        <div
          class="home__link-container home__link-container--groups"
        >
          Groups
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home translation Renders component with french translation 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/projects"
      >
        <div
          class="home__link-container home__link-container--projects"
        >
          Projets
        </div>
      </a>
      <a
        class="home__link"
        href="/datasets"
      >
        <div
          class="home__link-container home__link-container--datasets"
        >
          Datasets
        </div>
      </a>
      <a
        class="home__link"
        href="/teams"
      >
        <div
          class="home__link-container home__link-container--groups"
        >
          Groupes
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
